### PR TITLE
Feature/upload organization logo

### LINKED ISSUE #43: Add organization logo
Organizations should be able to add their logo. The logo will be displayed in Organization page and in Organization catalog.

# Definition of done
* [ ] Button for adding a logo exists in Add organization view.
* [x] There is a collection for storing the logo.
* [x] Logo is displayed on single organization page.

# Wireframe
In Add Organization form
addui

After uploading a logo
addui2

In card view (if implemented)
organizationcard

In organization view, the icon would appear like this:
organizationprofile3

### PR DESCRIPTION
**Pull request by jykae**
Closes #43

### CODE REVIEW

**Comment by jykae** [comment]
Waiting add organization view to be merged #49 to include upload logo.

File: organizations/logo/upload/client/resumable.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        // Update logo id field
+        Organizations.update(organization._id, { $set: { organizationLogoFileId } });
+
+        // Get success message translation
+        const message = TAPi18n.__('organizationLogo_resumable_successfully_uploaded');
+
+        sAlert.success(message);
+
+        return OrganizationLogo.resumable.upload();
```
Comment (marla-singer) [inline organizations/logo/upload/client/resumable.js]: It's very bad solution. You must update Organizations collection and show a message on fileSuccess event.
https://github.com/apinf/platform/blob/develop/branding/cover_photo/client/upload/resumable.js#L13-L28
In your case you add the id of non-existent image in Organization collection, tell to user that his image was updated and after then you upload the image.
I had the problems with this schema when network speed is slow. Don't do so :point_up:

**Comment by jykae** [comment]
#49 got merged, continuing..

**Comment by jykae** [comment]
Blocked, see comment https://github.com/Digipalvelutehdas/APIKA/issues/43#issuecomment-263560563

**Comment by brylie** [comment]
This should no longer be blocked.

/cc @bajiat

**Comment by jykae** [comment]
Continuing..

**Comment by jykae** [comment]
Looks now like this, hover shows camera icon, click opens file browse
When file is added, previous logo is removed & updated to use new uploaded logo.

Is circle style ok or should it be square ?

nayttokuva 2016-12-05 kello 16 32 37

**Comment by brylie** [comment]
> Is circle style ok or should it be square ?

We should not crop or block parts of the logo, as they are particular to organizations' branding. So, leave them at their original shape for now. We can come back to the decision at a later point.

**Comment by jykae** [comment]
@brylie Discussed quickly with @bajiat and switched back to square.

**Comment by brylie** [comment]
@jykae we should not cut or change the image aspect ratio. If possible, leave the image as it is, only resizing if necessary.

**Comment by jykae** [comment]
thanks @brylie for vertical align rule :)

@marla-singer ready for review

**Comment by marla-singer** [comment]
@jykae Cool, it works :) 
joxi_screenshot_1480952211813

Will logo changing be available in another task?

**Comment by jykae** [comment]
@marla-singer If you upload new image it removes previous logo, replaces with new upload. Is that what you mean logo changing?

**Comment by marla-singer** [comment]
@jykae If i want to delete current logo (in this case picture with cat) and upload another one.

**Comment by marla-singer** [comment]
@jykae I found some strange behavior. I'm testing and can't undesrtand when it happens

**Comment by marla-singer** [comment]
### Steps
1.Create a new organization
2. Upload logo
3. Pay attention that after logo uploading the url has **#** in the end (I used Chrome)
4. Click on the logo to change it.

### Found
File browser doesn't open. But

File: organizations/logo/view/client/view.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      if (organizationLogoFile) {
+        // Get API logo file URL
+        return `${Meteor.absoluteUrl().slice(0, -1) + OrganizationLogo.baseURL}/md5/${organizationLogoFile.md5}`;
+      }
+    }
+  },
+  organizationLogoExists () {
+    const organization = Template.currentData().organization;
+
+    if (organization && organization.organizationLogoFileId) {
```
Comment (marla-singer) [inline organizations/logo/view/client/view.js]: You can simplify this one as
```javascript
return organization && organization.organizationLogoFileId
```

**Comment by marla-singer** [comment]
@jykae Also it's needed to setting permission for uploading. Now non-owner can change organization logo

**Comment by jykae** [comment]
@marla-singer requested fixes done. Could not find way to get organization in OrganizationLogo collection permissions and can't possibly restrict to just admin, so added permission check to "fileAdded" event. Hope thats ok for now.

**Comment by marla-singer** [comment]
@jykae Your solution for permission works good as well :+1:

File: organizations/collection/helpers.js [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
+    // Check that user is logged in
+    if (userId) {
+      // Check if user is manager of this organization
+      const userIsManager = _.includes(this.managerIds, userId);
+
+      // Check if user is administrator
+      const userIsAdmin = Roles.userIsInRole(userId, ['admin']);
+
+      // if user is manager or administrator, they can edit
+      return userIsManager || userIsAdmin;
```
Comment (marla-singer) [inline organizations/collection/helpers.js]: And here you can check
```javascript
return this.createdBy === userId || userIsAdmin;
```

**Comment by jykae** [comment]
@marla-singer I would not want restrict organization managers to just one, if customer requirements are not clear & change. That way we can be more flexible and not need schema migrations. Imho usual decision in schema modeling. Organization manager is not necessarily the one who created organization.

**Comment by marla-singer** [comment]
@jykae Right, I see your point. So let's merge it

File: organizations/logo/view/client/view.html [hunk 4]
```diff
@@ -1,19 +1,16 @@
 <template name="viewOrganizationLogo">
+  <div class="profile-img-container organization-file-browse">
```
Comment (brylie) [inline organizations/logo/view/client/view.html]: Both of these classes look like they could be IDs. Consider removing one of them, or at least making one an ID for the div.

Comment (brylie) [inline organizations/logo/view/client/view.html]: E.g. the `profile-img-container` seems redundant, while the `organization-file-browser` seems like a good ID.

File: organizations/client/view/header/header.less [hunk 5]
```diff
@@ -27,31 +27,28 @@
     position: relative;
     display: inline-block;
     overflow: hidden;
+    cursor: pointer;
 }
 
 .profile-img-container img {
```
Comment (brylie) [inline organizations/client/view/header/header.less]: Try to remove CSS/LESS like this when possible. Rely instead on Bootstrap and semantic class names. Fewer lines of code is better.

File: organizations/logo/view/client/view.html (hunk 4 above)
Comment (brylie) [inline organizations/logo/view/client/view.html]: > The .img-responsive class applies max-width: 100%, height: auto, and display:block to the image.

File: organizations/client/view/header/header.less (hunk 5 above)
Comment (jykae) [inline organizations/client/view/header/header.less]: @brylie I don't know way to do this layer style without additional rules. But as discussed I move these rules to "view.less" so they are closer to place they are used.

File: organizations/logo/view/client/view.html (hunk 4 above)
Comment (jykae) [inline organizations/logo/view/client/view.html]: @brylie Good point about ID.

**Comment by jykae** [comment]
@brylie @marla-singer Done.

**Comment by marla-singer** [comment]
@brylie Is everything okay?

**Comment by brylie** [comment]
Jep, looks good. I was just wanting to reduce the number of custom classes and LESS code we maintain.

**Pull request merged**